src/libraries/System.Linq/src/System/Linq/Select.SpeedOpt.cs

```diff
@@ -165,9 +165,35 @@ public TResult TryGetLast(out bool found)
                 found = true;
                 return _selector(_source[_source.Length - 1]);
             }
+
+            bool ICollection<TResult>.Contains(TResult item)
```

Comment:
I can see why you think this could be controversial. I tend to think of "collections" as structures whose elements are all materialized in-memory, and any algorithmic complexity in their operations pertains to accessing said memory. In this case however both `Contains` and `CopyTo` methods entail arbitrary work (and side-effects!), which immediately makes me classify this as an "enumerable" and not a "collection". I suspect this can lead to very difficult to diagnose regressions and it's probably not worth the perf improvements.

Reply:
Ok, I've removed the second commit for Select